sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -93,38 +114,38 @@ public String getServiceBusResourceName() {
      */
     public Flux<ServiceBusReceivedMessage> receive() {
         if (isDisposed.get()) {
-            return Flux.error(new IllegalStateException("Cannot receive from a client that is already closed."));
+            return Flux.error(logger.logExceptionAsError(
+                new IllegalStateException("Cannot receive from a client that is already closed.")));
+        }
+
+        if (receiveMode != ReceiveMode.PEEK_LOCK && isAutoComplete) {
+            return Flux.error(logger.logExceptionAsError(new UnsupportedOperationException(
+                "Auto-complete is not supported on a receiver opened in ReceiveMode.RECEIVE_AND_DELETE.")));
```

Comment:
Autocomplete is a single word 😃 

```suggestion
                "Autocomplete is not supported on a receiver opened in ReceiveMode.RECEIVE_AND_DELETE.")));
```